### PR TITLE
Gutenframe: Move TinyMCE plugin to Calypso monorepo

### PR DESCRIPTION
**Pull request by mmtr**
Part of #32044.

#### Changes proposed in this Pull Request

* New `tinymce.js` file on the `@automattic/wpcom-block-editor` package containing the TinyMCE plugin that replaces the media modal used on the classic blocks with the Calypso media modal. It's the same logic that the one included on the `gutenberg-wpcom-classic-block-tinymce-plugin.js` file currently maintained in WP.com, but using ES2015+ syntax.
* New `utils.js` file on the `@automattic/wpcom-block-editor` package containing shared utilities.
* Added support for the `tinymce` dependency to the `@automattic/wordpress-external-dependencies-plugin` package, since it is available as global on WordPress environments (added @sirreal as reviewer to check this change).
* Added `tinymce` as dependency of the `@automattic/wpcom-block-editor` package so we don't need to declare it as global anymore. 
* Changed the build script so we generate an additional build bundle for the TinyMCE plugin.

#### Testing instructions

* Build the package: `npx lerna run build --scope='@automattic/wpcom-block-editor'`.
* Upload the `iframe-bridge-server.js` build file (`/apps/wpcom-block-editor/dist/iframe-bridge-server.js`) to your sandbox and overwrite the content of the `gutenberg-wpcom-iframe-bridge.js` file with the content of the build file.
* Upload the `tinymce.js` build file (`/apps/wpcom-block-editor/dist/tinymce.js`) to your sandbox and overwrite the content of the `gutenberg-wpcom-classic-block-tinymce-plugin.js` file with the content of the build file.
* Load `https://wordpress.com/block-editor` and select your sandbox site.
* Insert a classic block.
* Make sure the Calypso media modal appears for managing the media items added to the classic block.

### CODE REVIEW

File: apps/wpcom-block-editor/src/iframe-bridge-server.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* global tinymce, calypsoifyGutenberg */
+/* global calypsoifyGutenberg */
```
[the remark this answers is not in the document]
Comment (mmtr) [reply apps/wpcom-block-editor/src/iframe-bridge-server.js]: `calypsoifyGutenberg` is a global var defined by the Calypsoify plugin we load on the server (D20084-code). We use it for sharing some data from the server to the client.

File: packages/wordpress-external-dependencies-plugin/index.js [hunk 2]
```diff
@@ -34,6 +34,10 @@ class WordPressExternalDependenciesPlugin {
 				externRootRequest = 'ReactDOM';
 				break;
 
+			case 'tinymce/tinymce':
```
[the remark this answers is not in the document]
Comment (mmtr) [reply packages/wordpress-external-dependencies-plugin/index.js]: > Why `import 'tinymce/tinymce'` and not just `import 'tinymce'`? 

I guess this is caused by how the `tinymce` package is distributed. I checked out [their repo](https://github.com/tinymce/tinymce-dist) but I'm not sure where the `tinymce/tinymce` naming is coming from. In any case, it's [how we usually import it in Calypso](https://github.com/Automattic/wp-calypso/search?q=%22import+tinymce+from+%27tinymce%27%22&unscoped_q=%22import+tinymce+from+%27tinymce%27%22).

> I've been trying to track down this script and the handle that should be included… according to [this page](https://developer.wordpress.org/reference/functions/wp_enqueue_script/#default-scripts-included-and-registered-by-wordpress) perhaps it should be `tiny_mce`?

Do we really need to consider the handle? AFAIK, the handle is used to identify a script we enqueue on the server-side with `wp_enqueue_script` but it doesn't produce any result on the client-side. We only need to worry about the global vars this script declare so we can externalize the dependency on the package. And in this case, the Tiny MCE scripts defines a [`tinymce` global var](https://github.com/tinymce/tinymce-dist/blob/3a997f24eb54d890b1c88f22cff03092e0b73378/tinymce.js#L26822) we can use as a replacement of `import tinymce from 'tinymce/tinymce'`.

But yeah, maybe I'm missing something here so feel free to make any amendment you think we should do 🙂.

[the remark this answers is not in the document]
Comment (mmtr) [reply packages/wordpress-external-dependencies-plugin/index.js]: > I've been trying to track down this script and the handle that should be included… according to [this page](https://developer.wordpress.org/reference/functions/wp_enqueue_script/#default-scripts-included-and-registered-by-wordpress) perhaps it should be `tiny_mce`?

Ignore my response above. Just realized you were referring to the part of the logic that generates the `deps.json` files that should be used for indicating the dependencies of a script enqueued on WordPress.

And you were right, the handle is `tiny_mce`. Added in bc2bb32

File: packages/wordpress-external-dependencies-plugin/index.js [hunk 3]
```diff
@@ -67,15 +71,23 @@ class WordPressExternalDependenciesPlugin {
...
-									: userRequest
-							);
+							let handle;
+
+							if ( userRequest.startsWith( WORDPRESS_NAMESPACE ) ) {
+								// Transform @wordpress deps:
+								//   @wordpress/i18n -> wp-i18n
+								//   @wordpress/escape-html -> wp-escape-html
+								handle = 'wp-' + userRequest.substring( WORDPRESS_NAMESPACE.length );
+							} else if ( 'tinymce/tinymce' === userRequest ) {
```
Comment (kwight) [inline packages/wordpress-external-dependencies-plugin/index.js]: Why is this block needed? Can we not rename so that the final `else` works fine?..

**Comment by kwight** [comment]
Wonderful, confirmed I can see the new utils and tinymce code, and that the Calypso media modal loads from the Classic block as expected 👍

File: packages/wordpress-external-dependencies-plugin/index.js (hunk 3 above)
Comment (mmtr) [reply packages/wordpress-external-dependencies-plugin/index.js]: See my comment above: https://github.com/Automattic/wp-calypso/pull/32162#discussion_r274237905

**Pull request merged**